xcp/cpiofile.py

```diff
@@ -519,6 +519,10 @@ def seek(self, pos):
             self.init()
         self.read(pos - self.pos)
 
+    def init(self):
```

Comment:
I think this can be done via some of the @abc annotations in python, would https://docs.python.org/3/library/abc.html#abc.abstractmethod (and inheriting from the proper ABC class) fix the pylint warning too?

Reply:
Yes, using abc works just as well. That one is annoying in projects making heavy use of metaclasses (which is why I default to the exception idiom), but it will do the job here.

Reply:
Note, inheriting from `abc.ABC` comes with 3.4 only, so we'd have to use something like `@six.add_metaclass(abc.ABCMeta)` for a 2/3-portable idiom, which I feel brings too much complexity, for no clear advantage over `NotImplementedError()`.  What do you think ?

Reply:
> Note, inheriting from `abc.ABC` comes with 3.4 only, so we'd have to use something like `@six.add_metaclass(abc.ABCMeta)` for a 2/3-portable idiom, which I feel brings too much complexity, for no clear advantage over `NotImplementedError()`. What do you think ?

I think I'd concur with that. My feeling is we want the Py2/Py3 compatibility phase to be relatively short until the migration is complete and then drop the py2 support completely at which point there can be a follow on set of changes to perform code cleanups.